crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -3035,4 +3035,80 @@ mod timed_tests {
         assert_eq!(events3.len(), 1);
         assert_eq!(events3[0].event_id().as_deref(), Some(evid2));
     }
+
+    #[async_test]
+    async fn test_rfind_event_in_memory_by() {
```

Comment:
What I would like to see in this test is proof that it doesn't look inside the store; can you have a partially filled in-memory linked chunk, and look for an event that's in the store but not in memory, please?